src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1625,10 +1753,13 @@ private AuthenticationResult refreshToken(final CallbackHandler callbackHandle,
                 // remove item from cache to avoid same usage of
                 // refresh token in next acquireToken call
                 removeItemFromCache(refreshItem);
```

Comment:
We get to the point of removal when we know that it's expired. Otherwise you Cache keeps growing and filled with expired tokens...